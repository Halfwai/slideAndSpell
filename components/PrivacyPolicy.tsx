import { View, ScrollView, Linking, TouchableOpacity, StyleSheet } from 'react-native';
import MyAppText from '@/components/MyAppText';
import AuthButton from '@/components/AuthButton';
import { COLOURS } from '@/constants/colours';


export default function PrivacyPolicy(props: { hidePrivacyPolicy: Function }) {
    return (
        <View style={styles.privacyPolicyOuterContainer}>
            <View style={styles.privacyPolicyContainer}>
                <MyAppText style={styles.title}>Privacy Policy</MyAppText>
                <ScrollView style={styles.scrollView}>
                    
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        Effective Date: 7th March 2025
                    </MyAppText>
                    <MyAppText>                              
                        Slide and Spell respects your privacy. This Privacy Policy explains how we handle your information when you use our word game app Slide and Spell.
                    </MyAppText>
                    <MyAppText                                
                        style={styles.policyHeader}
                    >
                        Information We Collect
                    </MyAppText>
                    <MyAppText>
                        We collect and store your email address solely for authentication purposes. We do not collect, store, or process any other personal information.
                    </MyAppText>
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        How We Use Your Information
                    </MyAppText>
                    <MyAppText>
                        Your email address is used exclusively to authenticate your identity and allow access to the App. We do not share, sell, or use your email address for any other purpose.
                    </MyAppText>
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        Data Security
                    </MyAppText>
                    <MyAppText>
                        We implement appropriate security measures to protect your email address from unauthorized access, disclosure, or misuse.
                    </MyAppText>
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        Third-Party Services
                    </MyAppText>
                    <MyAppText>
                        Our App does not share your email address with third parties or integrate with third-party services for data processing.
                    </MyAppText>
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        Your Rights
                    </MyAppText>
                    <MyAppText>
                        Since we only collect your email address for authentication, you have the right to request its deletion. If you wish to delete your account and associated email address, please contact us at 
                        <TouchableOpacity 
                            style={{height: "100%"}}
                            onPress={() => Linking.openURL('mailto:slideandspell@gmail.com')}
                        >
                                <MyAppText
                                    style={styles.privacyPolicyText}
                                >slideandspell@gmail.com</MyAppText>
                        </TouchableOpacity>                            
                        .
                    </MyAppText>
                    <MyAppText                         
                        style={styles.policyHeader}
                    >
                        Changes to This Policy
                    </MyAppText>
                    <MyAppText>
                        We may update this Privacy Policy from time to time. Any changes will be posted within the App, and continued use of the App after changes are made signifies your acceptance of the updated policy.
                    </MyAppText>
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        Contact Us
                    </MyAppText>
                    <MyAppText>
                        If you have any questions about this Privacy Policy, please contact us at 
                        <TouchableOpacity 
                            style={{height: "100%"}}
                            onPress={() => Linking.openURL('mailto:slideandspell@gmail.com')}
                        >
                                <MyAppText
                                    style={styles.privacyPolicyText}
                                >slideandspell@gmail.com</MyAppText>
                        </TouchableOpacity>
                        .
                    </MyAppText>
                </ScrollView>
                <AuthButton text="Close" onPress={() => props.hidePrivacyPolicy()} style={{ backgroundColor: COLOURS.green }}/>
            </View>     
        </View>
    )
}

const styles = StyleSheet.create({
    title: {
        fontSize: 30,
    },
    privacyPolicyText: {
        textDecorationLine: 'underline',
    },
    input: {
        width: '90%',
        marginTop: 10,
        borderWidth: 1,
        borderRadius: 10,
        backgroundColor: 'white',
        padding: 10,
        fontSize: 20,
        height: 50,
    },
    privacyPolicyOuterContainer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    privacyPolicyContainer: {
        height: '80%',
        width: '80%',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: COLOURS.blue,
        borderRadius: 10,
        borderWidth: 1,
        borderColor: 'black',
    },
    scrollView: {
        width: '90%',
        backgroundColor: 'white',
        paddingHorizontal: 20,
        borderRadius: 10,
        margin: 10,
    },
    policyHeader: {
        fontSize: 20,
        fontWeight: 'bold',
    }
})