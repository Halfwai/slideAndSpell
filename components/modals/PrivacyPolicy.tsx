import React from 'react';
import { View, ScrollView, Linking, TouchableOpacity, StyleSheet } from 'react-native';
import { RFPercentage } from 'react-native-responsive-fontsize';

// Import COLOURS
import { COLOURS } from '@/constants/colours';

// Import components
import MyAppText from '@/components/common/MyAppText';
import AuthButton from '@/components/buttons/AuthButton';

// Setup props
interface PrivacyPolicyProps {
    hidePrivacyPolicy: Function
}

export default function PrivacyPolicy(props: PrivacyPolicyProps) {
    return (
        <View style={styles.privacyPolicyOuterContainer}>
            <View style={styles.privacyPolicyContainer}>
                <MyAppText style={styles.title}>Privacy Policy</MyAppText>
                <ScrollView style={styles.scrollView}>
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        Effective Date: 7th March 2025
                    </MyAppText>
                    <MyAppText style={styles.policyText}>
                        Slide and Spell respects your privacy. This Privacy Policy explains how we handle your information when you use our word game app Slide and Spell.
                    </MyAppText>
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        Information We Collect
                    </MyAppText>
                    <MyAppText style={styles.policyText}>
                        We collect and store your email address solely for authentication purposes. We do not collect, store, or process any other personal information.
                    </MyAppText>
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        How We Use Your Information
                    </MyAppText>
                    <MyAppText style={styles.policyText}>
                        Your email address is used exclusively to authenticate your identity and allow access to the App. We do not share, sell, or use your email address for any other purpose.
                    </MyAppText>
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        Data Security
                    </MyAppText>
                    <MyAppText style={styles.policyText}>
                        We implement appropriate security measures to protect your email address from unauthorized access, disclosure, or misuse.
                    </MyAppText>
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        Third-Party Services
                    </MyAppText>
                    <MyAppText style={styles.policyText}>
                        Our App does not share your email address with third parties or integrate with third-party services for data processing.
                    </MyAppText>
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        Your Rights
                    </MyAppText>
                    <MyAppText style={styles.policyText}>
                        Since we only collect your email address for authentication, you have the right to request its deletion. If you wish to delete your account and associated email address, please contact us at
                        <TouchableOpacity
                            onPress={() => Linking.openURL('mailto:slideandspell@gmail.com')}
                        >
                            <MyAppText
                                style={styles.privacyPolicyText}
                            >slideandspell@gmail.com</MyAppText>
                        </TouchableOpacity>
                        .
                    </MyAppText>
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        Changes to This Policy
                    </MyAppText>
                    <MyAppText style={styles.policyText}>
                        We may update this Privacy Policy from time to time. Any changes will be posted within the App, and continued use of the App after changes are made signifies your acceptance of the updated policy.
                    </MyAppText>
                    <MyAppText
                        style={styles.policyHeader}
                    >
                        Contact Us
                    </MyAppText>
                    <MyAppText style={styles.policyText}>
                        If you have any questions about this Privacy Policy, please contact us at
                        <TouchableOpacity
                            onPress={() => Linking.openURL('mailto:slideandspell@gmail.com')}
                        >
                            <MyAppText
                                style={styles.privacyPolicyText}
                            >slideandspell@gmail.com</MyAppText>
                        </TouchableOpacity>
                        .
                    </MyAppText>
                </ScrollView>
                <AuthButton text="Close" onPress={() => props.hidePrivacyPolicy()} style={{ backgroundColor: COLOURS.green }} />
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    title: {
        fontSize: RFPercentage(4),
    },
    privacyPolicyText: {
        textDecorationLine: 'underline',
        fontSize: RFPercentage(2.5),
    },
    input: {
        width: '90%',
        marginTop: 10,
        borderWidth: 1,
        borderRadius: 10,
        backgroundColor: 'white',
        padding: 10,
        fontSize: RFPercentage(2.5),
        height: 50,
    },
    privacyPolicyOuterContainer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    privacyPolicyContainer: {
        height: '80%',
        width: '80%',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: COLOURS.blue,
        borderRadius: 10,
        borderWidth: 1,
        borderColor: 'black',
    },
    scrollView: {
        width: '90%',
        backgroundColor: 'white',
        paddingHorizontal: 20,
        borderRadius: 10,
        margin: 10,
    },
    policyHeader: {
        fontSize: RFPercentage(3),
        fontWeight: 'bold',
    },
    policyText: {
        fontSize: RFPercentage(2.5),
    },
})