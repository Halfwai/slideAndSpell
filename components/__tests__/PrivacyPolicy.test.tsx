import { render } from '@testing-library/react-native';
import PrivacyPolicy from '@/components/modals/PrivacyPolicy';

it(`PrivacyPolicy Component renders correctly`, () => {
    const { toJSON } = render(
        <PrivacyPolicy
            hidePrivacyPolicy={() => {}}
        />
    );    
    expect(toJSON()).toMatchSnapshot();
});