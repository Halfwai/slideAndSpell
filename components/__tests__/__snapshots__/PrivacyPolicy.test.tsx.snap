// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyPolicy Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#CBE6F7",
        "borderColor": "black",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": "80%",
        "justifyContent": "center",
        "width": "80%",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          {
            "fontSize": 30,
          },
        ]
      }
    >
      Privacy Policy
    </Text>
    <RCTScrollView
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "margin": 10,
          "paddingHorizontal": 20,
          "width": "90%",
        }
      }
    >
      <View>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              {
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Effective Date: 7th March 2025
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              undefined,
            ]
          }
        >
          Slide and Spell respects your privacy. This Privacy Policy explains how we handle your information when you use our word game app Slide and Spell.
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              {
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Information We Collect
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              undefined,
            ]
          }
        >
          We collect and store your email address solely for authentication purposes. We do not collect, store, or process any other personal information.
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              {
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
        >
          How We Use Your Information
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              undefined,
            ]
          }
        >
          Your email address is used exclusively to authenticate your identity and allow access to the App. We do not share, sell, or use your email address for any other purpose.
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              {
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Data Security
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              undefined,
            ]
          }
        >
          We implement appropriate security measures to protect your email address from unauthorized access, disclosure, or misuse.
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              {
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Third-Party Services
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              undefined,
            ]
          }
        >
          Our App does not share your email address with third parties or integrate with third-party services for data processing.
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              {
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Your Rights
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              undefined,
            ]
          }
        >
          Since we only collect your email address for authentication, you have the right to request its deletion. If you wish to delete your account and associated email address, please contact us at
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "Quicksand",
                  },
                  {
                    "textDecorationLine": "underline",
                  },
                ]
              }
            >
              slideandspell@gmail.com
            </Text>
          </View>
          .
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              {
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Changes to This Policy
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              undefined,
            ]
          }
        >
          We may update this Privacy Policy from time to time. Any changes will be posted within the App, and continued use of the App after changes are made signifies your acceptance of the updated policy.
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              {
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Contact Us
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Quicksand",
              },
              undefined,
            ]
          }
        >
          If you have any questions about this Privacy Policy, please contact us at
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "Quicksand",
                  },
                  {
                    "textDecorationLine": "underline",
                  },
                ]
              }
            >
              slideandspell@gmail.com
            </Text>
          </View>
          .
        </Text>
      </View>
    </RCTScrollView>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#8BF0BC",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginBottom": 20,
          "opacity": 1,
          "padding": 12,
          "width": "90%",
        }
      }
      testID="authButton"
    >
      <Text
        style={
          [
            {
              "fontFamily": "Quicksand",
            },
            {
              "color": "black",
              "fontSize": 20,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Close
      </Text>
    </View>
  </View>
</View>
`;
